src/ng/browser.js

```diff
@@ -235,10 +235,22 @@ function Browser(window, document, $log, $sniffer) {
 
   // This variable should be used *only* inside the cacheState function.
   var lastCachedState = null;
+  function getCurrentState() {
+    try {
+      var state = history.state;
+      if (state === void 0) {
```

Comment:
Perhaps we should change `isDefined()` to use this method of testing for `undefined` and use it here?